### PR TITLE
Following #6

### PR DESCRIPTION
**Pull request by NelsonFrancisco**
Props to @timhuff 
I'm only following [this PR](https://github.com/griffinpp/objection-soft-delete/pull/6) and adding tests and documentation.

Test it for yourselves :)

### CODE REVIEW

**Comment by timhuff** [comment]
Thanks for following up on this work, @NelsonFrancisco . I kind of dropped the ball on following through with that PR. Good to see it'll finally come through.

**Comment by griffinpp** [comment]
Awesome work, I will take a closer look at this today

File: tests.js [hunk 1]
```diff
@@ -166,6 +174,38 @@ describe('Soft Delete plugin tests', () => {
           });
       });
     });
+
+    describe('when deletedValue and nonDeletedValue are overridden', () => {
+      before(() => {
+        return knex.schema.table('TestObjects', (table) => {
```
Comment (NelsonFrancisco) [inline tests.js]: I tried to
```js
before(createDeletedAtColumn(knex))
after(removeDeletedAtColumn(knex))
```
And then, at the beggining of the file, outside the first `describe` scope, declare `createDeletedAtColumn` and `removeDeletedAtColumn` but it didn't work. 
It says that knex is undefined. javascript shenanigans I'd say. Anyway, open to suggestions

**Comment by NelsonFrancisco** [comment]
> It would be good to put the column creation code into separate functions for future use, as was your initial instinct. Otherwise, this looks great and I'll be happy to merge once that bit is done. My apologies for taking so long. Life has been throwing me quite a few curveballs these days...

no problem. 
I've taken your ideas into account and updated it. 
Turns out I just needed to call the functions inside a function inside `before` and `after` so knex is defined

**Pull request merged**